### PR TITLE
added Schema and ERD

### PR DESCRIPTION
**Pull request by popeinabar**
this pull request consists of schema and ERD diagram for Customer-Relationship-Management and here is the [video 1](https://www.loom.com/share/f2b9305ca55e4d05809550a90da67e2d?sid=63dec994-8f42-43ff-86d2-ef07a711a46a) and [video 2](https://www.loom.com/share/049c8e9f03f441e6bb37aa1488241607?sid=d188e797-5326-4223-9691-a4561b47ff1e) where I am explaining schema and ERD

[video](https://www.loom.com/share/db5710380ed44baba5ca184886ad0ec6?sid=58e2cf7b-a683-493b-ad9e-599eaab6e72b) for milestone 2

### CODE REVIEW

**Comment by th3c0d3br34ker** [comment]
Please schedule a viva with me.

**Review by th3c0d3br34ker — CHANGES_REQUESTED** [review]
thanks @popeinabar, as per our discussion during the viva, please do the required changes and let me know.

**Comment by popeinabar** [comment]
I updated the schema and erd based on the feedback provided. The commit can be found above.

File: Screenshot 2023-09-29 at 12.52.15 PM.png
Comment (th3c0d3br34ker) [inline Screenshot 2023-09-29 at 12.52.15 PM.png:1]: image


The the relationship between `Roles` and `Product` should be between `User` and `Product`.

**Review by th3c0d3br34ker — CHANGES_REQUESTED** [review]
thanks @popeinabar, as per the viva taken, there are still some changes which need to be done. please go through the comments and resolve them.

**Comment by th3c0d3br34ker** [comment]
hey @popeinabar please consider merging this and raise a separate PR for `DDL commands`.

please consider re-submitting with the correct links.

**Comment by th3c0d3br34ker** [comment]
please schedule a viva as well, as per my observation, this submission ignores our previous discussion done during the viva.

**Comment by popeinabar** [comment]
The ERD is updated according to our previous conversation.

**Comment by th3c0d3br34ker** [comment]
hey @popeinabar, please merge this PR for `ERD` and `Schema`. as per discussion during viva.

**Review by th3c0d3br34ker — APPROVED** [review]

**Pull request merged**